### PR TITLE
[software-bridges 1/x] Update API to support automatic bridge creation

### PR DESCRIPTION
**Pull request by ykulazhenkov**
Update API to support automatic bridge creation

SriovNetworkNodePolicySpec - Bridge field added
SriovNetworkNodeStateSpec - Bridges field added
SriovNetworkNodeStateStatus - Bridges field added
VirtualFunction - RepresentroName field added

Update API according to the [design doc](https://github.com/k8snetworkplumbingwg/sriov-network-operator/blob/master/doc/design/software-bridge-management.md)

### CODE REVIEW

File: api/v1/sriovnetworknodestate_types.go [hunk 1]
```diff
@@ -71,22 +72,48 @@ type InterfaceExt struct {
...
+	// bridge-level configuration for the bridge
+	Bridge OVSBridgeConfig `json:"bridge,omitempty"`
+	// uplink-level bridge configuration for each uplink(PF).
+	// currently must contain only one element
+	Uplinks []OVSUplinkConfigExt `json:"uplinks,omitempty"`
+}
+
+type OVSUplinkConfigExt struct {
+	// pci address of the PF
 	PciAddress string `json:"pciAddress"`
```
Comment (SchSeba) [inline api/v1/sriovnetworknodestate_types.go:106]: will this always exist even if a bond is connected to the ovs?

**Review by SchSeba — COMMENTED** [review]
small nits and we can merge this first part

File: api/v1/sriovnetworknodestate_types.go (hunk 1 above)
Comment (ykulazhenkov) [reply api/v1/sriovnetworknodestate_types.go:106]: yes, in this case we will have two or more entries in the OVSCofnigExt.uplinks, each entry represents a single interface

File: api/v1/sriovnetworknodepolicy_types.go [hunk 2]
```diff
@@ -62,6 +62,9 @@ type SriovNetworkNodePolicySpec struct {
 	ExcludeTopology bool `json:"excludeTopology,omitempty"`
 	// don't create the virtual function only allocated them to the device plugin. Defaults to false.
 	ExternallyManaged bool `json:"externallyManaged,omitempty"`
+	// contains bridge configuration for matching PF
```
[the remark this answers is not in the document]
Comment (ykulazhenkov) [reply api/v1/sriovnetworknodepolicy_types.go]: done

**Review by zeeke — COMMENTED** [review]
LGTM
please, resolve code conflicts

File: api/v1/sriovnetworknodepolicy_types.go [hunk 3]
```diff
@@ -77,6 +80,53 @@ type SriovNetworkNicSelector struct {
...
+type OVSUplinkConfig struct {
+	// contains settings for PF interface in the OVS bridge
+	Interface OVSInterfaceConfig `json:"interface,omitempty"`
+}
+
+// OVSInterfaceConfig contains some options from the Interface table of the OVSDB for PF
+type OVSInterfaceConfig struct {
+	// type field in the Interface table in OVSDB
+	Type string `json:"type,omitempty"`
+	// type filed in the Interface table in OVSDB
```
Comment (adrianchiris) [inline api/v1/sriovnetworknodepolicy_types.go:122]: nit:
type -> options
filed -> field

File: api/v1/sriovnetworknodestate_types.go [hunk 4]
```diff
@@ -71,22 +72,48 @@ type InterfaceExt struct {
...
+	// name of the bridge
+	Name string `json:"name"`
+	// bridge-level configuration for the bridge
+	Bridge OVSBridgeConfig `json:"bridge,omitempty"`
+	// uplink-level bridge configuration for each uplink(PF).
+	// currently must contain only one element
+	Uplinks []OVSUplinkConfigExt `json:"uplinks,omitempty"`
+}
+
+type OVSUplinkConfigExt struct {
```
Comment (adrianchiris) [inline api/v1/sriovnetworknodestate_types.go:104]: nit: add docstring

File: api/v1/sriovnetworknodestate_types.go [hunk 5]
```diff
@@ -71,22 +72,48 @@ type InterfaceExt struct {
...
+	VdpaType        string `json:"vdpaType,omitempty"`
+	RepresentorName string `json:"representorName,omitempty"`
+}
+
+// Bridges contains list of bridges
+type Bridges struct {
+	OVS []OVSConfigExt `json:"ovs,omitempty"`
+}
+
+type OVSConfigExt struct {
```
Comment (adrianchiris) [inline api/v1/sriovnetworknodestate_types.go:94]: nit: add docstring

**Review by adrianchiris — COMMENTED** [review]
a few small nits otherwise LGTM !

File: api/v1/sriovnetworknodepolicy_types.go (hunk 3 above)
Comment (ykulazhenkov) [reply api/v1/sriovnetworknodepolicy_types.go]: done

File: api/v1/sriovnetworknodestate_types.go (hunk 4 above)
Comment (ykulazhenkov) [reply api/v1/sriovnetworknodestate_types.go:106]: done

File: api/v1/sriovnetworknodestate_types.go (hunk 5 above)
Comment (ykulazhenkov) [reply api/v1/sriovnetworknodestate_types.go:95]: done

**Comment by ykulazhenkov** [comment]
@zeeke @adrianchiris Thanks for the review. I addressed your comments.

**Review by zeeke — APPROVED** [review]

**Review by SchSeba — APPROVED** [review]
Nice work!

**Review by adrianchiris — APPROVED** [review]

**Pull request merged**